### ISSUE TITLE
Ogg embedded covers support

### ISSUE THREAD

**Issue by Oleksiy-Yakovenko** [DeaDBeeF-Player/deadbeef#985]
Original [issue 1083](https://code.google.com/p/ddb/issues/detail?id=1083) created by Alexey-Yakovenko on 2014-03-20T20:53:09.000Z:

Ogg containers use a tag called METADATA_BLOCK_PICTURE to store a base64 encoded structure containing an image.  For Flac, the albumart plugin reads this tag as cover art, using th Flac library, but for other Ogg codecs it doesn't.

I can create a patch to just have the plugin examine more file types in the same way, but possibly it is better to have decoder plugins supply relevant tag data in some defined fashion?  In metadata?  As a callback for each plugin?

**Comment by tmose1106** [comment]
Has there been any progress with this 'enhancement'? I have been fiddling for a few hours trying to embed album art in my OGG Vorbis audio files, and I have come to the relization that DeadBeef doesn't load the album artwork in the METADATA_BLOCK_PICTURE tag. Meanwhile, Poweramp on my Android device loads this tag just fine. I love DeadBeef, but this seems kinda ridiculous.

**Comment by Alexey-Yakovenko** [comment]
@tmose1106 
This issue is about the Desktop version of DeaDBeeF, which doesn't use the same codebase as Android.

The Android version doesn't support reading embedded album art from any file formats.

It will start supporting the same album art features, as desktop version, in version 2.0.

The new album art plugin, which would be compatible with android, is under development right now.

**Comment by tmose1106** [comment]
Sorry for any confusion. What I'm saying is that Deadbeef on the desktop, with the current git master, does not load the METADATA_BLOCK_PICTURE tag on OGG Vorbis files. It simply lists it as a metadata tag. 

Other audio players are able to read this tag just fine. So there is some functionality missing in DeadBeef's embedded art reader.

**Comment by Alexey-Yakovenko** [comment]
@tmose1106 
Yes, this is correct, this issue is open specifically to keep track of this missing feature.

As I mentioned above, right now I'm actively developing a new album art plugin, which will hopefully replace the old one soon (in version 0.8), and I'll make sure it supports more file formats than the old one.

**Comment by tmose1106** [comment]
Alright, thank you. I think I can make due until then.

**Comment by Blurrit** [comment]
Has there been any progress on this?
As of version 1.8.0, DeaDBeeF still does not load album art for opus or ogg files.

I have found a dirty hack to make it work for opus files, namely to change the file extension to mp3, add album art using EasyTAG, and then change the extension back to opus. But I don't think it's a good solution...

**Comment by Alexey-Yakovenko** [comment]
Unfortunately, I could not fit any of the album art work into 1.8 scope. So no, it's still not done.

**Comment by richharp825** [comment]
Update: ogg album art now works for ogg files, since version 1.9.4. Album art for opus files still does not work.

**Comment by richharp825** [comment]
As of DeaDBeeF 1.10.1, album art works correctly for both ogg and opus files. The fix happened probably in https://github.com/DeaDBeeF-Player/deadbeef/pull/3192. I believe this issue can now be closed.
